lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -371,26 +372,27 @@ sub _extended_needle_info {
 
 sub src {
     my ($self) = @_;
-
     return $self->reply->not_found unless $self->init();
 
     my $job    = $self->stash('job');
     my $module = $self->stash('module');
 
+    if (my $casedir = $job->settings->find({key => 'CASEDIR'})) {
+        my $casedir_url = Mojo::URL->new($casedir->value);
+        # if CASEDIR points to a remote location let's assume it is a git repo
+        # that we can reference like gitlab/github
+        last unless $casedir_url->scheme;
+        $casedir_url->path($casedir_url->path . '/blob/' . $casedir_url->fragment . '/' . $module->script);
+        $casedir_url->fragment('');
+        return $self->redirect_to($casedir_url);
+    }
     my $testcasedir = testcasedir($job->DISTRI, $job->VERSION);
     my $scriptpath  = "$testcasedir/" . $module->script;
-    if (!$scriptpath || !-e $scriptpath) {
-        $scriptpath ||= "";
-        return $self->reply->not_found;
-    }
+    return $self->reply->not_found unless $scriptpath && -e $scriptpath;
     my $script_h = path($scriptpath)->open('<:encoding(UTF-8)');
-    my @script_content;
-    if (defined $script_h) {
-        @script_content = <$script_h>;
-    }
-    my $script = "@script_content";
-
-    $self->stash('script',     $script);
+    return $self->reply->not_found unless defined $script_h;
+    my @script_content = <$script_h>;
+    $self->stash('script',     "@script_content");
     $self->stash('scriptpath', $scriptpath);
```

Comment:
I'd like to see a render call here (`$self->render(script => ..., scriptpath => ...)`). Implicit rendering doesn't belong in a complex app like openQA. I know, not your fault, but you can fix it here.

Reply:
Ok, cool. This is actually something that confused me that the rendering is implicit :D

For this and the refactoring I did before implementing the new feature I had created https://github.com/os-autoinst/openQA/pull/2747 . This issue fixed in https://github.com/os-autoinst/openQA/pull/2747/files#diff-f3aa6c205f4dc5f9c671808173d2eb59R385